### ISSUE TITLE
subsequent republishing attempt failed with 2015-10-12 23:11:36,704 [ERROR  ] local variable 'out' referenced before assignment (UnboundLocalError) (main.py:206)

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#262]
```
% datalad publish-collection ssh://datalad.org/srv/collections.datalad.org/www/demo demo_collection
2015-10-12 23:11:36,511 [ERROR  ] DEBUG: ssh -o "ControlMaster=yes" -o "ControlPath=/var/run/user/47521/datalad/datalad.org" -o "ControlPersist=yes" datalad.org exit (publish_collection.py:153)
bind: Address already in use
channel_setup_fwd_listener_tcpip: cannot listen to port: 3389
Could not request local forwarding.
2015-10-12 23:11:36,703 [ERROR  ] Failed to run "ssh -S /var/run/user/47521/datalad/datalad.org datalad.org 'cat | sh /dev/stdin' /srv/collections.datalad.org/www/demo DATALAD_COL_demo_collection demo_handle2 demo_handle1 < /home/yoh/proj/datalad/datalad-debian/datalad/resources/sshserver_prepare_for_publish.sh" under None. Exit code=1. out=## start server capabilities ####
| DATALAD_GIT_VERSION: git version 2.1.4 DATALAD_END
| DATALAD_GIT_ANNEX_VERSION: git-annex version: 5.20141125
| build flags: Assistant Webapp Webapp-secure Pairing Testsuite S3 WebDAV Inotify DBus DesktopNotify XMPP DNS Feeds Quvi TDFA CryptoHash
| key/value backends: SHA256E SHA1E SHA512E SHA224E SHA384E SKEIN256E SKEIN512E SHA256 SHA1 SHA512 SHA224 SHA384 SKEIN256 SKEIN512 WORM URL
| remote types: git gcrypt S3 bup directory rsync web webdav tahoe glacier ddar hook external DATALAD_END
| ## end server capabilities ####
| error: target 'DATALAD_COL_demo_collection' exists
|  err= (cmd.py:264)
2015-10-12 23:11:36,703 [ERROR  ] Preparation script failed: CommandError: command 'ssh -S /var/run/user/47521/datalad/datalad.org datalad.org 'cat | sh /dev/stdin' /srv/collections.datalad.org/www/demo DATALAD_COL_demo_collection demo_handle2 demo_handle1 < /home/yoh/proj/datalad/datalad-debian/datalad/resources/sshserver_prepare_for_publish.sh' failed with exitcode 1.
| Failed to run "ssh -S /var/run/user/47521/datalad/datalad.org datalad.org 'cat | sh /dev/stdin' /srv/collections.datalad.org/www/demo DATALAD_COL_demo_collection demo_handle2 demo_handle1 < /home/yoh/proj/datalad/datalad-debian/datalad/resources/sshserver_prepare_for_publish.sh" under None. Exit code=1. out=## start server capabilities ####
| DATALAD_GIT_VERSION: git version 2.1.4 DATALAD_END
| DATALAD_GIT_ANNEX_VERSION: git-annex version: 5.20141125
| build flags: Assistant Webapp Webapp-secure Pairing Testsuite S3 WebDAV Inotify DBus DesktopNotify XMPP DNS Feeds Quvi TDFA CryptoHash
| key/value backends: SHA256E SHA1E SHA512E SHA224E SHA384E SKEIN256E SKEIN512E SHA256 SHA1 SHA512 SHA224 SHA384 SKEIN256 SKEIN512 WORM URL
| remote types: git gcrypt S3 bup directory rsync web webdav tahoe glacier ddar hook external DATALAD_END
| ## end server capabilities ####
| error: target 'DATALAD_COL_demo_collection' exists
|  err= (publish_collection.py:172)
2015-10-12 23:11:36,704 [ERROR  ] local variable 'out' referenced before assignment (UnboundLocalError) (main.py:206)
```

**Comment by bpoldrack** [comment]
Missed to get stdout, stderr for evaluation in case of non-zero exit of these scripts. The non-zero is caused by the already existing dirs in this case. Fixed in PR #265

**Comment by mih** [comment]
Was reported "fixed" long time ago.
